### ISSUE TITLE
🚀 Feature: re-write LlamaIndex instrumentation to use LlamaIndex `CallbackManager`

### ISSUE THREAD

**Issue by nirga** [traceloop/openllmetry-js#134]
### Which component is this feature for?

LlamaIndex Instrumentation

### 🔖 Feature description

Right now, we monkey-patch classes and methods in LlamaIndex which requires endless work and constant maintenance. LlamaIndex has a system for [callbacks](https://ts.llamaindex.ai/api/classes/CallbackManager) that can potentially be used to create/end spans without being too coupled with with the framework's inner structure.

### 🎤 Why is this feature needed ?

Support LlamaIndex entirely and be future-proof to internal API changes

### ✌️ How do you aim to achieve this?

Look into LlamaIndex `callback_manager` and how other frameworks are using it.

### 👀 Have you spent some time to check if this feature request has been raised before?

- [X] I checked and didn't find similar issue

### Are you willing to submit PR?

None

**Comment by kartikay-bagla** [comment]
@quest-bot embark

**Comment by kartikay-bagla** [comment]
@nirga I was looking at the [LlamaDebugHandler (in the python code)](https://docs.llamaindex.ai/en/stable/examples/callbacks/LlamaDebugHandler.html) and it looks like in terms of features, I just need to recreate this and extend it to add the existing traces. Does that sound correct?

**Comment by nirga** [comment]
Yep @kartikay-bagla. It's a bit trickier cause you need to create and close spans accordingly, and make sure to nest them correctly.

**Comment by kartikay-bagla** [comment]
Small update, with this being a larger code change than expected, and some unscheduled life things, I might have to put this on hold until the 18th. Will keep you updated though.

**Comment by kartikay-bagla** [comment]
So I was able to make some progress over the weekend, but it is slow as I couldn't find any existing implementation of callback_manager that I could use as a starting point.

**Comment by nirga** [comment]
Thanks so much @kartikay-bagla! I've been in touch with the folks at LlamaIndex so ping me here or on slack if you have any questions I can pass along.

**Comment by JCaraballo113** [comment]
Hi, I will try to solve this issue. I do have one question. This feature is desired for the JS SDK or the Python SDK or both?
